pageserver/src/deletion_queue/list_writer.rs

```diff
@@ -243,21 +236,19 @@ impl ListWriter {
             let file_name = dentry.file_name();
             let dentry_str = file_name.to_string_lossy();
 
-            if Some(file_name.as_os_str()) == header_path.file_name() {
+            if file_name.to_str() == header_path.file_name() {
```

Comment:
Unsure about this, I think there should be a PartialEq for OsStr and str?

Reply:
Perhaps all read_dir users need to be go over after this PR, no need to change yet.

Reply:
This was in [955c1c7](https://github.com/neondatabase/neon/pull/5352/commits/955c1c728ead0cbfce4ec4b52a516a3f2e486681).

It's probably good right now.